internal/brokers/manager.go

```diff
@@ -183,3 +183,10 @@ func (m *Manager) BrokerForSessionID(sessionID string) (broker *Broker, err erro
 
 	return broker, nil
 }
+
+// UnsetBrokerForSessionID removes any previously selected broker for a given transaction sessionID.
+func (m *Manager) UnsetBrokerForSessionID(sessionID string) {
```

Comment:
Do you think there would be a way to clean it up from the broker itself (without keeping a dangling reference to the manager?).

Also, thinking about it, once IsAuthorized returns success or denied, we need to clean it up from the manager too.

Reply:
I don't think so... This is the broker manager, so it has control of the brokers but they don't have access to it, so I think we can't really leave this up to the broker itself.

Reply:
What if we don’t go to the same roundtrip for this one and only let the manager cancelling a broker transaction by sessionID?

Reply:
I'm not sure what you mean. This function is only called by the PAM service, AFAIK. When [asked to abort a session](https://github.com/ubuntu/authd/blob/5f9acaac192e5f76cd9781a4446045f34b841bba/internal/services/pam/pam.go#L193), the service also should remove the `session -> broker` mapping from the broker manager as well. It's not a dangling reference, since the PAM service already has a `BrokerManager` self-contained in it.

Reply:
Oh, I see it now... Yeah, it makes sense and it's doable for sure. I only have a single question about it, then: Doesn't it get too dissonant from the other methods in the service API? Even though it is a dance, it's a "familiar" dance (like a "I feel like I've done this before and know how it goes" situation).

I'm not opposed to changing it, it's just a "design" question.

Reply:
I think for cancelling a whole transaction is like "removing a broker for blablabla", so I feel this can be the duty of the manager itself to handle the whole transaction creation/drop. I agree that’s debatable, but the code will be simpler that way for the same amount of additional public API.

Reply:
I've pushed the changes so we can see how it will look.
TBH, I didn't find it very attractive, but I'll trust your judgment.